Default-enable `llvm_tools_enabled` when no `config.toml` is present

jieyouxu opened the pull request:
Fixes #133195. cc @wesleywiser could you double check if with this patch and no `config.toml` that you can run `./x test tests/ui --stage 1`?

`llvm-objcopy` is usually required by cg_ssa on macOS to workaround bad `strip`s.

cc @bjorn3 I hope this doesn't break cg_clif...

r? bootstrap

jieyouxu committed:
Default-enable `llvm_tools_enabled` when no `config.toml` is present

jieyouxu committed:
Register change info

bjorn3 commented:
> I hope this doesn't break cg_clif...

Does this disable the option? https://github.com/rust-lang/rust/blob/e6c1e14e5d487a90abeb5ba4dccc608521bbfb54/compiler/rustc_codegen_cranelift/scripts/setup_rust_fork.sh#L32-L33 If not, it either should, or the config file there will need to be changed to explicitly disable it.

Kobzol commented on src/bootstrap/src/core/config/config.rs:
```diff
@@ -1255,6 +1255,10 @@ impl Config {
 
+            // This is needed by codegen_ssa on macOS to ship `llvm-objcopy` aliased to
+            // `rust-objcopy` to workaround bad `strip`s on macOS.
+            llvm_tools_enabled: true,
```
Funny, `config.example.toml` already claimed that it was enabled by default before.

Kobzol commented on src/bootstrap/src/utils/change_tracker.rs:
```diff
@@ -300,4 +300,9 @@ pub const CONFIG_CHANGE_HISTORY: &[ChangeInfo] = &[
+    ChangeInfo {
+        change_id: 133207,
+        severity: ChangeSeverity::Info,
+        summary: "llvm tools are is now enabled by default when no `config.toml` is provided.",
```
```suggestion
        summary: "`rust.llvm-tools` is now enabled by default when no `config.toml` is provided.",
```

jieyouxu replied:
Fixed.

jieyouxu committed:
Explicitly disable llvm tools for cranelift

jieyouxu commented:
> If not, it either should, or the config file there will need to be changed to explicitly disable it.

I modified the config there to explicitly disable llvm tools.

jieyouxu commented:
Changes since last review ([diff](https://github.com/rust-lang/rust/compare/2c39c2bf97573e7cb17ed7de2e1e4a205ee46a3b..70592623a1f527ed1cb422cec8b3ea3810593fbc)):

- Fixed change log entry
- Adjusted cg_clif's config.toml to disable llvm tools

jieyouxu commented on compiler/rustc_codegen_cranelift/scripts/setup_rust_fork.sh:
```diff
@@ -38,6 +38,9 @@ local-rebuild = true
 codegen-backends = ["cranelift"]
 deny-warnings = false
 verbose-tests = false
+# Enabled by default, but cranelift does not need/want llvm tools.
```
Fixed comment

Kobzol commented:
LGTM, if bjorn3 is ok with the latest change.

jieyouxu commented:
@bors r=Kobzol,bjorn3 rollup

bors commented:
:pushpin: Commit 4b5c88301bcc20de9fa58b7390d78d596f7d7223 has been approved by `Kobzol,bjorn3`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Kobzol,bjorn3 4b5c88301bcc20de9fa58b7390d78d596f7d7223 -->
<!-- homu: {"type":"Approved","sha":"4b5c88301bcc20de9fa58b7390d78d596f7d7223","approver":"Kobzol,bjorn3","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

wesleywiser reviewed (approved):
I can confirm this fixes building without `config.toml` for me. Thanks @jieyouxu! 💛
